test/unified_format.py

```diff
@@ -1804,14 +1818,28 @@ def check_events(self, spec):
             else:
                 assert server_connection_id is None
 
+    def process_ignore_messages(self, ignore_logs, actual_logs):
+        final_logs = []
+        for log in actual_logs:
```

Comment:
NIT: Thoughts on this potential optimization?


actual_log_messages = {actual_log["data"]["message"] for actual_log in actual_logs}
return list(filter(lambda log: log["data"]["message"] not in actual_log_messages))

Reply:
I like this optimization but it does deviate from the spec's statement that ["Matching rules used to match messages in `ignoreMessages` are identical to match rules used for `messages` matching".](https://github.com/mongodb/specifications/blob/a35de873621a244b88cad01aa3cd7c8da31a5ecd/source/unified-test-format/unified-test-format.md?plain=1#L1235) Is this a case where we need to stick closely to the spec, or can we do this much simpler way of matching?

Reply:
Good point. I'll say let's not then. 

The `self.match_evaluator.match_result()` function could be added in, but we lose readability so we'll keep it as is.